src/ophys_etl/utils/traces.py

```diff
@@ -1,11 +1,30 @@
 from typing import List
-from functools import partial
 import numpy as np
 from scipy.sparse import coo_matrix
 from scipy.ndimage.filters import median_filter
 
-# Partial for simplifying repeat median filter calls
-medfilt = partial(median_filter, mode='constant')
+
+def nanmedian_filter(x: np.ndarray, filter_length: int) -> np.array:
+    """ 1D median filtering with np.nanmedian
+    Parameters
+    ----------
+    x: np.ndarray
+        1d array of signal
+    filter_length: int
+        Length of the median filter to compute a rolling baseline
+
+    Return
+    ------
+    filtered_trace
+    """
+    half_length = int(filter_length/2)
+    # Create 'reflect' traces at the extrema
+    temp_trace = np.concatenate(
+        (np.flip(x[:half_length]), x, np.flip(x[-half_length:])))
+    filtered_trace = np.zeros_like(x)
+    for i in range(len(x)):
+        filtered_trace[i] = np.nanmedian(temp_trace[i:i+filter_length])
```

Comment:
Also, one additional thing: How does the code handle the case where where the slice for an index is all NaN values?

Reply:
I created a conditional to fill in a slice with all NaN values with the median of the previous index.